source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -0,0 +1,152 @@
+﻿//#define COSMOSDEBUG
+
+using System;
+using System.Collections.Generic;
+using Cosmos.HAL.BlockDevice;
+using Cosmos.System.FileSystem.VFS;
+
+namespace Cosmos.System.FileSystem
+{
+    public class Disk
+    {
+        private List<ManagedPartition> parts = new List<ManagedPartition>();
+        private static List<FileSystemFactory> registeredFileSystems = new List<FileSystemFactory>();
+        private PartitioningType partitioningType;
+
+        public PartitioningType PartitioningType { get { return partitioningType; } }
+        /// <summary>
+        /// The size of the disk in MB.
+        /// </summary>
+        public int Size { get; }
+        /// <summary>
+        /// List of partitions
+        /// </summary>
+        public List<ManagedPartition> Partitions { get { return parts; } }
```

Comment:
Is the list necessary here? Cant we access the information from PartitioningType already? Also if we create a new Partition via the PartitioningType how do we update the list here?